### PR TITLE
Swap sqlite libraries

### PR DESCRIPTION
**Pull request by SuperQ**
Switch to a SQLite library that doesn't require CGO.

Fixes: https://github.com/prometheus/promlens/issues/40

Signed-off-by: SuperQ <superq@gmail.com>

### CODE REVIEW

**Comment by SuperQ** [comment]
I tested using both combinations. The sqlite3 mode with an sqlite file, and a sqlite mode with the sqlite3 file. 

Seems like the file format works between the two libraries.
```
$ file /tmp/promlens-links.db*
/tmp/promlens-links.db:  SQLite 3.x database, last written using SQLite version 3039002, file counter 5, database pages 4, cookie 0x2, schema 4, UTF-8, version-valid-for 5
/tmp/promlens-links.db3: SQLite 3.x database, last written using SQLite version 3039002, file counter 3, database pages 4, cookie 0x2, schema 4, UTF-8, version-valid-for 3
```

File: README.md [hunk 1]
```diff
@@ -75,7 +75,7 @@ To run PromLens with link sharing and Grafana integration enabled, using SQLite
 
 ```
 ./promlens \
-  --shared-links.sql.driver=sqlite3 \
+  --shared-links.sql.driver=sqlite \
```
Comment (juliusv) [inline README.md:78]: We just need to remember to list this as a breaking change in the changelog.

**Comment by juliusv** [comment]
:+1: Nice! Before I merge, just wanted to confirm that I understood you correctly that you tried writing some shared links with the old library, then you confirmed that the new library was able to read those, right?

**Comment by Kickball** [comment]
For What Its Worth:
I haven't tested a migration, but from a fresh start, a binary built from this commit successfully writes to the SQLite database and persists the links.

**Comment by SuperQ** [comment]
@juliusv Yes, I tested read/write both ways. Maybe it's good enough that we can make the change transparent and allow `sqlite3` for backwards compatibility.

File: pkg/sharer/sharer.go [hunk 2]
```diff
@@ -168,7 +168,7 @@ func NewSQLSharer(logger log.Logger, driver string, dsn string, createTables boo
 			}
 		}
 
-	case "sqlite3":
+	case "sqlite":
```
Comment (SuperQ) [inline pkg/sharer/sharer.go:171]: @juliusv What do you think of this:

```suggestion
	case "sqlite3", "sqlite":
```

Comment (juliusv) [reply pkg/sharer/sharer.go:171]: Yeah, let's do that, great!

Comment (SuperQ) [reply pkg/sharer/sharer.go:171]: Done, took a slightly different tactic, but will also nicely log a warning now.

**Comment by juliusv** [comment]
:+1: Nice!

**Pull request merged**